packages/flutter/lib/src/storage/core_store_directory_io.dart

```diff
@@ -1,11 +1,71 @@
-import 'package:path_provider/path_provider.dart';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:path_provider/path_provider.dart' as path_provider;
+import 'package:path/path.dart' as path;
 
 class CoreStoreDirectory {
   Future<String> getDatabaseDirectory() async {
-    return (await getApplicationDocumentsDirectory()).path;
+    if (defaultTargetPlatform == TargetPlatform.iOS) {
+      await _migrateDBFileToLibraryDirectory();
+
+      return (await path_provider.getLibraryDirectory()).path;
+    }
+
+    return (await path_provider.getApplicationDocumentsDirectory()).path;
+  }
+
+  /// Why do we need this migration on iOS? see:
+  /// https://github.com/parse-community/Parse-SDK-Flutter/issues/791
+  ///
+  /// TODO: remove this migration algorithm after at least two major releases
+  ///
+  /// the release when this migration added is `3.1.14`
+  Future<void> _migrateDBFileToLibraryDirectory() async {
+    final dbFile = await _getDBFileIfExistsInAppDocDir();
+
+    if (dbFile != null) {
+      await _moveDatabaseFileToLibraryDirectory(dbFile);
+    }
+  }
+
+  Future<File?> _getDBFileIfExistsInAppDocDir() async {
+    final appDocDirPath =
+        (await path_provider.getApplicationDocumentsDirectory()).path;
+
+    final databaseFilePath = path.join(
+      appDocDirPath,
+      'parse',
+      'parse.db',
+    );
+
+    final dbFile = File(databaseFilePath);
+
+    if (await dbFile.exists()) {
+      return dbFile;
+    }
+
+    return null;
+  }
+
+  Future<void> _moveDatabaseFileToLibraryDirectory(
+    File databaseFileToMove,
+  ) async {
+    final libraryDirectoryPath =
+        (await path_provider.getLibraryDirectory()).path;
+
+    final libraryDirectoryDatabaseFilePath = path.join(
+      libraryDirectoryPath,
+      'parse',
+      'parse.db',
+    );
+
+    await File(libraryDirectoryDatabaseFilePath).create(recursive: true);
+    await databaseFileToMove.copy(libraryDirectoryDatabaseFilePath);
```

Comment:
Instead of copy + delete, is there a move command? In some file system that simply changes the file pointer instead of actually duplicating the file and then deleting the original.

Reply:
Thanks for pointing this out. 
Yes, there is [rename()](https://api.dart.dev/stable/2.19.2/dart-io/File/rename.html) function that will do the trick